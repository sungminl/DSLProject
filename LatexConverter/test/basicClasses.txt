TEXT(Bold, Heading, Underline, Italics){"abcdefg this is my text entry"}
&a = 3.4
&x = a + 42
&b = "abc"
&c = [1, 2, 3]
&d = ["a", "b", "c"]

GRAPH{
NODES:{Node1(S), Node2(A)},
CONNECTIONS:{Node1->(text)Node2}
}

TABLE 
ROW:3,COL:3
TOPHEADER: ("T1", "T2", "T3")
LEFTHEADER: ("L1", "L2", "L3")
{
"a", &b, "c"|
"a", "b", "c"|
"a", "b", "c"|
}

MATRIX 
ROW:3,COL:3 (&a)
{
2, 2.54, -5|
&a, &a + 1, 5|
5 ^ (42 + x), log(5), a * 2|
}

/* this is the text inside the 
commented code
*/