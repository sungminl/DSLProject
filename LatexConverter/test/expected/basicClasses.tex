\documentclass[]{article}
\usepackage{amsmath}
\usepackage{tikz}
\usetikzlibrary{graphs,automata}
\title{Document}
\begin{document}
\section*{\textbf{\textit{\underline{abcdefg this is my text entry}}}}

\begin{tikzpicture}[node distance={2cm}, thick, state/.style = {draw, circle}]
\node[state,initial left] (Node1) {Node1};
\node[state,accepting] (Node2) [right of=Node1] {Node2};
\path[->] (Node1) edge[bend right = 30] node[below = 0.1cm] {text} (Node2);
\end{tikzpicture}

\begin{center}
\begin{tabular}{|l|c|c|c|}
\hline
&\textbf{T1}&\textbf{T2}&\textbf{T3}\\
\hline
\textbf{L1}&a&abc&c\\
\hline
\textbf{L2}&a&b&c\\
\hline
\textbf{L3}&a&b&c\\
\hline
\end{tabular}
\end{center}

\[\begin{bmatrix}
2.0 & 2.54 & -5.0\\
3.4 & 4.4 & 5.0\\
5.0 ^ { 42.0 + ( x ) } &  \log( 5.0 ) & a \cdot ( 2.0 )\\
\end{bmatrix}\]
\end{document}
